streams/src/main/java/org/apache/kafka/streams/processor/TopologyBuilder.java

```diff
@@ -1000,6 +1007,8 @@ public synchronized Pattern sourceTopicPattern() {
     }
 
     public synchronized void updateSubscriptions(SubscriptionUpdates subscriptionUpdates) {
```

Comment:
Ak. 

 I had an integration test, but it was reading from topics.  I was thinking a better test would be to pull the `ProcessorTopology` out and verify all source nodes are properly set.  Does that seem reasonable?